### ISSUE TITLE
"Top of body" hook?

### ISSUE THREAD

**Issue by zerocrates** [omeka/omeka-s#988]
Kim's user bar change is in, and currently depends on themes to add a call to its partial in their layout.

At the time, we determined that would be easiest because there were no events/triggers/etc. there, so either option would have required changes to all the themes.

Do we want an event like that (for adding things at the top of the body?) The theme side can be trivially changed to work that way if we do, but we just need to decide yes/no before propagating the changes down to all the themes.

**Comment by zerocrates** [comment]
Ignore the parts of this that talk about not propagating the changes.

**Comment by patrickmj** [comment]
I'm generally in favor of adding an event at places where it might be useful, and this looks like a pretty useful place. Using that event to add the language switching some people are looking for, for example, makes sense to me.

**Comment by zerocrates** [comment]
You mean the interface for that switching? It actually might make sense for us to have a hook that hooks into the bar itself, as well.

**Comment by zerocrates** [comment]
Looking at the merges on the themes reminds me that a straight up event might be problematic for this as we currently need to do some trickery to get the bar's assets loaded in the page head which would be tougher in an event.

**Comment by patrickmj** [comment]
Yeah, I was thinking of the interface for that switching. So a hook into the bar itself would make sense, but it might be weird to have to handle both the logged-in and logged-out situations for that.
